webapp/src/components/blockIconSelector.test.tsx

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {ReactElement} from 'react'
+import {fireEvent, render, screen} from '@testing-library/react'
+
+import userEvent from '@testing-library/user-event'
+
+import '@testing-library/jest-dom'
+import {IntlProvider} from 'react-intl'
+
+import {BlockIcons} from '../blockIcons'
+import {TestBlockFactory} from '../test/testBlockFactory'
+
+import BlockIconSelector from './blockIconSelector'
+
+const wrapIntl = (children: ReactElement) => <IntlProvider locale='en'>{children}</IntlProvider>
+const board = TestBlockFactory.createBoard()
+const icon = BlockIcons.shared.randomIcon()
+
+jest.mock('../mutator', () => ({
+    changeIcon: jest.fn((id, oldIcon, newIcon) => {
+        board.fields.icon = newIcon
+        return Promise.resolve()
+    }),
+}))
+
+describe('components/blockIconSelector', () => {
+    beforeEach(() => {
+        board.fields.icon = icon
+    })
+    test('return an icon correctly', () => {
+        render(wrapIntl(
+            <BlockIconSelector
+                block={board}
+                size='l'
+            />,
+        ))
+        expect(screen.queryByText(icon)).toBeTruthy()
+    })
+    test('return no element with no icon', () => {
+        board.fields.icon = ''
+        const {container} = render(wrapIntl(
+            <BlockIconSelector
+                block={board}
+                size='l'
+            />,
+        ))
+        expect(container).toBeEmptyDOMElement()
+    })
+    test('return menu on click', () => {
+        const {container} = render(wrapIntl(
+            <BlockIconSelector
+                block={board}
+                size='l'
+            />,
+        ))
+        userEvent.click(container.getElementsByTagName('div')[1]) //0 is div "BlockIconSelector"
```

Comment:
This looks a bit suspicious because it relies on the tags and their order.

May be we can start adding [ARIA roles](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques#roles) to the elements and use them for testing purposes also. For example [this `div` inside `MenuWrapper`](https://github.com/mattermost/focalboard/blob/main/webapp/src/widgets/menuWrapper.tsx#L84-L88) can be given a `button` role.

Reply:
I agree with you for the aria role

When I wrote the test, I started with the idea of leaving the existing code alone and writing the tests around that code.
But if I can add elements in the existing code then yes, I want to avoid that 😃 

Just modify the MenuWrapper by adding this aria-label and role. Can i do it? 😃

Reply:
I think that you can do such things when writing unit tests. Look for example at some recent pull request with unit test that [added data-test-id](https://github.com/mattermost/focalboard/pull/906/files#diff-2e274a77dd6b2957a2f7e0579fa989e0cbfca3656a62cc5a49c772ee09aabf89R35)

Reply:
Ok, I add data-testid on menuwrapper ;)